tensorflow/core/kernels/sequence_ops.cc

```diff
@@ -91,18 +91,26 @@ class RangeOp : public OpKernel {
           errors::InvalidArgument(
               "Requires start >= limit when delta < 0: ", start, "/", limit));
     }
-    int64_t size;
+    uint64_t size;
     if constexpr (std::is_integral<T>::value) {
-      size = Eigen::divup(Eigen::numext::abs(limit - start),
-                          Eigen::numext::abs(delta));
+      uint64_t range;
+      if ((limit > 0 && start < 0) || (limit < 0 && start > 0)) {
+        range = static_cast<uint64_t>(Eigen::numext::abs(limit)) 
+                + static_cast<uint64_t>(Eigen::numext::abs(start));
+      } else {
+        range = static_cast<uint64_t>(Eigen::numext::abs(limit - start)); 
+      }
+
+      size = Eigen::divup(range,
+                          static_cast<uint64_t>(Eigen::numext::abs(delta)));
```

Comment:
@mihaimaruseac I think we should add a check here that `size` is less than or equal to `std::numeric_limits<int64_t>::max()`.